onnxruntime/core/providers/cpu/rnn/deep_cpu_lstm.cc

```diff
@@ -874,7 +884,7 @@ void UniDirectionalLstm<T>::Compute(const gsl::span<const T>& inputs_arg,
       }
     };
 
-    ExecuteLambdaInParallel("Processing batch", hidden_gemm_and_activations, batch_size_, fused_hidden_rows, ttp_, logger_);
+    ExecuteLambdaInParallel("Processing batch", hidden_gemm_and_activations, batch_size_, fused_hidden_rows, lstm_tp_, logger_);
 
```

Comment:
If we use the mlas_tp_ here, and pass nullptr in the calls to ComputeGemm on line 836 would that mean we don't need a threadpool in each LSTM node?

Reply:
Yes. Are you suggesting removing the lstm_tp?

Reply:
Yes. Let mlas provide the threading at the ExecuteLambdaInParallel level, and single thread the GEMM call within the lambda

---
In reply to: [313455794](https://github.com/microsoft/onnxruntime/pull/1609#discussion_r313455794) [](ancestors = 313455794)

Reply:
I'd prefer to put it into the next PR.